[BUG] Referencing different Meta Model versions causes unexpected exception

**Describe the bug**
When validating an Aspect Model which references multiple versions of the Meta Model, the validation fails with an unexpected error.

**Where**
The exception occurs in the Aspect Model Validator.

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88)
Caused by: io.vavr.MatchError: type: java.util.NoSuchElementException, value: java.util.NoSuchElementException: No value present
        at io.vavr.API$Match.of(API.java:5095)
        at io.openmanufacturing.sds.aspectmodel.validation.services.AspectModelValidator.lambda$validate$6(AspectModelValidator.java:131)
        at io.vavr.control.Try.lambda$recover$6ea7267f$1(Try.java:949)
        at io.vavr.control.Try.of(Try.java:75)
        at io.vavr.control.Try.recover(Try.java:949)
        at io.openmanufacturing.sds.aspectmodel.validation.services.AspectModelValidator.validate(AspectModelValidator.java:131)
        at io.openmanufacturing.sds.BammCli.performModelValidation(BammCli.java:303)
        at io.openmanufacturing.sds.BammCli.run(BammCli.java:144)
        at io.openmanufacturing.sds.BammCli.main(BammCli.java:159)
        ... 8 more
```

**Additional context**
The same Aspect Model can be validated successfully when updating all Meta Model references to the latest version.

The issue can be reproduced using the latest state of the SDS SDK main branch and the attached Aspect Models: [Test Aspect.zip](https://github.com/OpenManufacturingPlatform/sds-sdk/files/8107990/Test.Aspect.zip)

--- a/core/sds-aspect-model-resolver/src/main/java/io/openmanufacturing/sds/aspectmodel/resolver/AspectModelResolver.java
+++ b/core/sds-aspect-model-resolver/src/main/java/io/openmanufacturing/sds/aspectmodel/resolver/AspectModelResolver.java
@@ -132,15 +132,12 @@ public class AspectModelResolver {
                   migratorService.getSdsMigratorFactory()
                                  .createAspectMetaModelResourceResolver()
                                  .mergeMetaModelIntoRawModel( model, oldest )
-                                 .orElse( migratorService.getMigratorFactory()
-                                                         .map( MigratorFactory::createAspectMetaModelResourceResolver )
-                                                         .map( Try::success )
-                                                         .orElseThrow()
-                                                         .flatMap( metaResolver -> metaResolver
-                                                               .mergeMetaModelIntoRawModel( model,
-                                                                     oldest ) ) )
-                                 .flatMap(
-                                       migratorService::updateMetaModelVersion ) ) );
+                                 .orElse( () -> migratorService.getMigratorFactory()
+                                                               .map( MigratorFactory::createAspectMetaModelResourceResolver )
+                                                               .map( Try::success )
+                                                               .orElseThrow()
+                                                               .flatMap( metaResolver -> metaResolver.mergeMetaModelIntoRawModel( model, oldest ) ) )
+                                 .flatMap( migratorService::updateMetaModelVersion ) ) );
    }
 
    /**

